src/generate/rust.rs

```diff
@@ -512,11 +505,9 @@ macro_rules! thunk {
     ($($t:tt)*) => {{
         let prefix = quote!($($t)*);
         Thunk::new(move |mut cont| {
-            // HACK(eddyb) remove awkward scope post-NLL
-            {
-                let code = cont.to_inline();
-                *code += mem::replace(code, prefix);
-            }
+            let code = cont.to_inline();
+            let old_code = mem::replace(code, prefix);
+            *code += old_code;
```

Comment:
Does `*code += mem::replace(code, prefix)` not work?

Reply:
```
error[E0499]: cannot borrow `*code` as mutable more than once at a time
   --> src/generate/rust.rs:509:35
    |
509 |             *code += mem::replace(code, prefix);
    |             ----------------------^^^^---------
    |             |                     |
    |             |                     second mutable borrow occurs here
    |             first mutable borrow occurs here
    |             first borrow later used here
...
771 |                     + opt(thunk!(assert_eq!(_range.start(), c.fn_input.start());)
    |                           ------------------------------------------------------- in this macro invocation
```

Reply:
That's... suboptimal, ugh. I guess you can rename `old_code` to `suffix`.